Source/Core/VideoBackends/D3D12/Render.cpp

```diff
@@ -1244,6 +1242,8 @@ void Renderer::SetSamplerState(int stage, int tex_index, bool custom_tex)
 	if (tex_index)
 		stage += 4;
 
+	const auto original_hex = gx_state.sampler[stage].hex;
```

Comment:
@lioncash `struct TexMode0` and `struct TexMode1` from VideoCommon/BPMemory.h are the actual hardware bit patterns; `SamplerState` is just an internal structure of the D3D12 backend that is populated with a local clone of some fields copied from the hardware state. The purpose of `SetSamplerState` seems to be to convert the current emulated hardware state (`bpmem` global) into an arbitrary internal format.

 Currently, the D3D12 backend does not store the sampler's associated anisotropy level in its `SamplerState` like the D3D11 one does for some reason. I'm pretty sure this is another bug, one that would require resizing `SamplerState` to `u64` to make room for the extra field since the current one is exactly 32bits wide. (The D3D11 `SamplerState` is already 64bits wide, I don't know why that was changed since a lot of this code was clearly copy-pasted)

I'm not sure what you mean by union/struct bitfield combo and replacing it with accessors? I know the BitField class is not quite a POD which messes up assignment because the operators are deleted, but raw C bitfields are not portable since the ordering is platform dependent.